controller/controller_server/utils.py

```diff
@@ -273,12 +386,15 @@ def get_snapshot_id_info(snapshot_id):
 def get_object_id_info(full_object_id, object_type):
     logger.debug("getting {0} info for id : {1}".format(object_type, full_object_id))
     splitted_object_id = full_object_id.split(config.PARAMETERS_OBJECT_ID_DELIMITER)
-    if len(splitted_object_id) != 2:
+    secret_uid = None
+    if len(splitted_object_id) == 2:
+        array_type, object_id = splitted_object_id
+    elif len(splitted_object_id) == 3:
+        array_type, secret_uid, object_id = splitted_object_id
+    else:
         raise ObjectIdError(object_type, full_object_id)
-
-    array_type, object_id = splitted_object_id
     logger.debug("volume id : {0}, array type :{1}".format(object_id, array_type))
-    return array_type, object_id
+    return array_type, object_id, secret_uid
```

Comment:
returning `secret_uid` in the middle would be more aligned with the order of appearances in the string

Reply:
done

Reply:
same for the order of the dataclass members (i.e. constructor parameters order here)

Reply:
done